Python/pywarpx/Bucket.py

```diff
@@ -59,7 +59,9 @@ def attrlist(self):
         for attr, value in self.argvattrs.items():
             if value is None:
                 continue
-            # --- repr is applied to value so that for floats, all of the digits are included.
+            # --- str (and repr) crop some digits for floats. TODO: we should format
+            #     floating point numbers & numpy scalars to the significant digits of the
+            #     precision of amrex::Real/ParticleReal
```

Comment:
I don't think that this would be needed. The values will be truncated anyway when read in. Though, a question to @WeiqunZhang is what happens when an input parameter has more digits than the type being read into? Is it rounded or truncated?

Reply:
On the AMReX/C++ side, extra digits above double will most certainly just be dropped.

Reply:
In principle, we need to format it have `std::numeric_limits<T>::max_digits10` digits. For `double` that's 17. Python's `str` seems to only print 16 digits.

In AMReX, we use `std::sstream` to read numbers. It's significant up to `std::numeric_limits<T>::max_digits10`.

Reply:
I think str gives all the digits. I've tested this `x - float(f'{x}')` on random numbers and it always gives zero.

Reply:
```
        std::cout << "digits10: " << std::numeric_limits<double>::digits10 << '\n'
                  << "max_digits10: " << std::numeric_limits<double>::max_digits10 << '\n';
        std::string str("1.2345678901234567890");
        for (int n = 15; n <= 17; ++n) {
            std::stringstream ss(str.substr(0,n+1)); // +2 because of `.`.                                                                                                   
            double num;
            ss >> num;
            std::cout << "With " << n << " digits, num == 1.2345678901234567890: " << (num == 1.2345678901234567890) << '\n';
        }
```
produces
```
digits10: 15
max_digits10: 17
With 15 digits, num == 1.2345678901234567890: 0
With 16 digits, num == 1.2345678901234567890: 0
With 17 digits, num == 1.2345678901234567890: 1
```

Reply:
What said about python was wrong. I happened to use a number whose 17th digit is small.
```
In [8]: str(1.23456789012345631234)
Out[8]: '1.2345678901234562'
```
It does print 17 digits.

Reply:
```
>>> x = np.float64(1.2345678901234567890)
>>> x - float(f'{x}')
np.float64(0.0)
```